osquery/tables/yara/yara_utils.cpp

```diff
@@ -382,6 +384,23 @@ Status YARAConfigParserPlugin::update(const std::string& source,
     }
   }
 
+  if (yara_config.HasMember("signature_urls")) {
+    auto& sigurl = yara_config["signature_urls"];
+    if (!sigurl.IsArray()) {
+      VLOG(1) << "YARA signature_url must be an array";
+    } else {
+      VLOG(1) << "Compiling YARA signature_url for allowed list";
+      for (const auto& element : sigurl.GetArray()) {
+        try {
```

Comment:
I am not sure this `try` is going to work as expected. I do not think RapidJSON throws if you try to access data of the wrong type. I think you need to first check `IsString()` before `GetString()`.

Reply:
The `try/catch` block is there to `catch` the exception thrown by `Uri`. It checks if the strings passed as signature URLs are valid or not and add them to the allowed set.